packages/user-ui/src/components/header/mod.rs

```diff
@@ -1,13 +1,267 @@
+                    }
+                    div { class: "flex flex-row w-full gap-[10px]",
+                        CustomCheckbox {
+                            lang,
+                            checked: checked_2(),
+                            onchange: move |v| {
+                                checked_2.set(v);
+                            },
+                        }
+                        div { class: "font-medium text-[#555462] text-[16px]", "{tr.agree_2}" }
+                        SeeDetailButton { lang }
+                    }
+                    if check_error() != "" {
+                        div { class: "font-normal text-red-400 text-[14px]", {check_error()} }
+                    }
+                }
+            }
+            div {
+                class: "cursor-pointer flex flex-row w-full h-[57px] justify-center items-center rounded-[12px] bg-[#8095EA] font-extrabold text-[18px] text-white",
+                onclick: move |_| {
+                    let email = email.clone();
+                    async move {
+                        if nickname() == "" {
+                            nickname_error.set(tr.nickname_error.to_string());
+                            return;
+                        } else if !checked_1() || !checked_2() {
+                            check_error.set(tr.check_error.to_string());
+                            return;
+                        }
+                        nickname_error.set("".to_string());
+                        check_error.set("".to_string());
+                        match user_service.login_or_signup(&email, &nickname()).await {
+                            Ok(_) => {
+                                popup_service
+                                    .open(rsx! {
+                                        CompletePopup {
+                                            lang,
+                                            onclose: move |_| {
+                                                popup_service.close();
+                                            },
+                                        }
+                                    })
+                                    .with_id("complete")
+                                    .with_title("VOICE KOREA");
+                            }
+                            Err(e) => {
+                                tracing::error!("signup failed: {:?}", e);
+                            }
+                        };
+                    }
+                },
+                "{tr.next}"
+            }
+        }
+    }
+}
+
+#[component]
+pub fn GoogleLoginPopup(lang: Language, onclose: EventHandler<MouseEvent>) -> Element {
+    let mut user_service: UserService = use_context();
+    let mut popup_service: PopupService = use_context();
+    let tr: GoogleLoginPopupTranslate = translate(&lang);
+
+    rsx! {
+        div { class: "flex flex-col min-w-[420px] justify-between items-center",
```

Comment:
min-w-[420px] will break mobile UI because most of mobile viewports are les than 400px. 
To cover mobile UI, you must consider mobile screen viewports less than 350px for all components.